feature-libs/product/future-stock/core/services/future-stock.service.ts

```diff
@@ -16,14 +20,17 @@ import { ProductFutureStock } from '../model';
 export class FutureStockService implements FutureStockFacade {
   protected readonly PRODUCT_KEY = 'productCode';
 
-  protected futureStockState$: Observable<ProductFutureStock> =
+  protected futureStockState$: Observable<ProductFutureStock | undefined> =
     this.routingService.getRouterState().pipe(
       withLatestFrom(this.userIdService.takeUserId()),
       switchMap(([{ state }, userId]) => {
-        return this.futureStockConnector.getFutureStock(
-          userId,
-          state.params[this.PRODUCT_KEY]
-        );
+        if (userId === OCC_USER_ID_CURRENT) {
```

Comment:
let's change the condition to 

`userId !== OCC_USER_ID_ANONYMOUS`

reason being, it will not always be current. It may be a customer emulation user, which is customerId I believe.
Also, if customer changes the endpoint to not use current, but the actual userId to make the request, then they won't be able to see the future stocks too.